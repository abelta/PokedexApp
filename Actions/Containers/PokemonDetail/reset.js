import * as types from '../../actionTypes';

const reset = () => ({ type: types.POKEMON_DETAIL_RESET });

export default reset;
