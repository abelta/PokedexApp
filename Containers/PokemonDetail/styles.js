import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  pokemonDetail: {
    flex: 1,
  },
  card: {
    flex: 1,
  },
});

export default styles;
