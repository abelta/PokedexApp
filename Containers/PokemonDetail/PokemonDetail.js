import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import { connect } from 'react-redux';
import pokemonShape, { pokemonDetailParams } from '../../PropTypes/Pokemon';
import { reset } from '../../Actions/Containers/PokemonDetail';
import { getSent as pokemonGetSent } from '../../Actions/Pokemon';
import { selectPokemon } from '../../Selectors/Pokemon';
import { ActivityIndicator, ErrorView, SlideView } from '../../Components';
import { Card as PokemonCard } from '../../Components/Pokemon';
import styles from './styles';

class PokemonDetail extends Component {
  componentWillMount() {
    const { getPokemon, match: { params: { id: name } }, pokemon, resetContainer } = this.props;
    resetContainer();
    if (!pokemon) getPokemon(name);
  }

  render() {
    const { error, loading, pokemon } = this.props;
    return (
      <View style={styles.pokemonDetail}>
        {loading && <ActivityIndicator />}
        {error && <ErrorView>{error}</ErrorView>}
        {
          pokemon &&
          <SlideView style={{ flex: 1 }} toLeft>
            <PokemonCard pokemon={pokemon} style={styles.card} />
          </SlideView>
        }
      </View>
    );
  }
}

PokemonDetail.propTypes = {
  error: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]).isRequired,
  getPokemon: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  match: pokemonDetailParams.isRequired,
  pokemon: PropTypes.oneOfType([pokemonShape, PropTypes.bool]).isRequired,
  resetContainer: PropTypes.func.isRequired,
};

const mapStateToProps = (state, props) => ({
  error: state.containers.pokemonDetail.error,
  loading: state.containers.pokemonDetail.loading,
  pokemon: selectPokemon(state, props.match.params.id),
});

const mapDispatchToProps = dispatch => ({
  getPokemon: name => dispatch(pokemonGetSent({ name })),
  resetContainer: () => dispatch(reset()),
});

export default connect(mapStateToProps, mapDispatchToProps)(PokemonDetail);
