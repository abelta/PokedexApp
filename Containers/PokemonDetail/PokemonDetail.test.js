import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import camelcaseKeys from 'camelcase-keys';
import PokemonDetail from '.';
import pokemonData from './mock-data-pokemon.json';

jest.mock('../../Components/SlideView');

const middlewares = [];
const mockStore = configureStore(middlewares);
const initialState = {
  containers: {
    pokemonDetail: {
      error: false,
      loading: false,
    },
    pokemonList: {
      error: false,
      loading: false,
      offset: 0,
    },
  },
  entities: {
    pokemon: {
      pikachu: camelcaseKeys(pokemonData, { deep: true }),
    },
    pokemonIndex: {},
  },
};
const store = mockStore(initialState);

describe('PokemonDetail', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <PokemonDetail match={{ params: { id: 'pikachu' } }} />
    </Provider>,
  );
  it('renders properly', () => {
    expect(tree).toMatchSnapshot();
  });
});
