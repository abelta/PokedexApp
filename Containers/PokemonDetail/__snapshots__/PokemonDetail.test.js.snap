// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonDetail renders properly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
